battlemap_processor/core/duplicate_detector.py

```
@@ -0,0 +1,450 @@
+
+                return image_hash
+            else:
+                return None
+
+        except Exception as e:
+            logger.warning(f"Error hashing image {image_info.filename}: {e}")
+            return None
+
+    def _calculate_file_hash(self, file_path: str) -> str:
+        """Calculate MD5 hash of a file"""
+        hash_md5 = hashlib.md5()
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)